envoy/network/dns.h

```diff
@@ -35,6 +35,25 @@ class ActiveDnsQuery {
    * @param reason supplies the cancel reason.
    */
   virtual void cancel(CancelReason reason) PURE;
+  /** Add a trace for the DNS query. */
+  void addTrace(uint8_t trace) {
+    absl::MutexLock lock(&lock_);
+    traces_.emplace_back(trace);
+  }
+  /** Return the DNS query traces. */
+  std::vector<uint8_t> getTraces() {
+    absl::MutexLock lock(&lock_);
+    return traces_;
+  }
+  /** Remove the DNS query traces. */
+  void removeTraces() {
+    absl::MutexLock lock(&lock_);
+    traces_.clear();
+  }
+
+private:
+  absl::Mutex lock_;
+  std::vector<uint8_t> traces_ ABSL_GUARDED_BY(lock_);
```

Comment:
what do these individual values mean in the vector?  shouldn't we enumerate them by name to make it clear what each value represents?